File: net/schedule-expiration/AddFilter.cs
// DocSection: schedule_expiration_add_filter
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Delivery;

// Creates an instance of the delivery client
// ProTip: Use DI for this in your apps https://docs.kontent.ai/net-register-client
IDeliveryClient client = DeliveryClientBuilder
      .WithProjectId("8d20758c-d74c-4f59-ae04-ee928c0816b7")
      .Build();

// Gets all landing pages
// Create strongly typed models according to https://docs.kontent.ai/strongly-typed-models
var response = await client.GetItemsAsync<LandingPage>(
    new EqualsFilter("system.type", "landing_page")
);

var today = System.DateTime.Today;

// Filters the landing pages, keeping those that should be public
var itemsToDisplay = response.Items.Where((item) =>
		(item.ExpireAt > today || item.ExpireAt == null));

return View(itemsToDisplay);
// EndDocSection

File: android/schedule-expiration/AddFilter.java
// DocSection: schedule_expiration_add_filter
// Tip: Find more about JavaRx SDK at https://docs.kontent.ai/androidandroid

// Prepares the DeliveryOptions configuration object
DeliveryOptions options = DeliveryOptions.builder()
.projectId("8d20758c-d74c-4f59-ae04-ee928c0816b7")
.build();

// Initializes a DeliveryClient for Java projects
DeliveryClient client = new DeliveryClient(options);

// Registers the model class for landing pages
client.registerType(LandingPage.class);

CompletionStage<List<LandingPage>> publishedItems = client.getItems(
                LandingPage.class,
                DeliveryParameterBuilder.params()
                        .filterEquals("system.type", "landing_page")
                        .build()
        ).thenApply(landingPageItems -> landingPageItems.stream()
                .filter(item ->
                        (item.getExpireAt().compareTo(ZonedDateTime.now()) > 0 || item.getExpireAt() == null))
                .collect(Collectors.toList()));


// Gets all landing pages using RxJava
Observable.fromCompletionStage(publishedItems)
    .subscribe(new Observer<List<LandingPage>>() {
        @Override
        public void onSubscribe(Disposable d) {
        }

        @Override
        public void onNext(List<LandingPage> landingPages) {
            // Already landing pages, that should be public
            List<LandingPage> publishedItems = landingPages;
        }

        @Override
        public void onError(Throwable e) {
        }

        @Override
        public void onComplete() {
        }
    });
// EndDocSection